simtools/corsika/corsika_output.py

```diff
@@ -0,0 +1,1301 @@
+            if False, the histograms are filled for all given telescopes together.
+            if True, one histogram is set for each telescope separately.
+        """
+        self.individual_telescopes = individual_telescopes
+        self.num_of_hist = len(self.telescope_indices) if self.individual_telescopes is True else 1
+
+        self.hist_position, self.hist_direction, self.hist_time_altitude = [], [], []
+
+        for _ in range(self.num_of_hist):
+            boost_axes_position = self._create_regular_axes("hist_position")
+            self.hist_position.append(
+                bh.Histogram(
+                    boost_axes_position[0],
+                    boost_axes_position[1],
+                    boost_axes_position[2],
+                )
+            )
+            boost_axes_direction = self._create_regular_axes("hist_direction")
+            self.hist_direction.append(
+                bh.Histogram(
+                    boost_axes_direction[0],
+                    boost_axes_direction[1],
+                )
+            )
+
+            boost_axes_time_altitude = self._create_regular_axes("hist_time_altitude")
+            self.hist_time_altitude.append(
+                bh.Histogram(
+                    boost_axes_time_altitude[0],
+                    boost_axes_time_altitude[1],
+                )
+            )
+
+    def _fill_histograms(self, photons, azimuth_angle=None, zenith_angle=None):
+        """Fill all the histograms created by self._create_histogram with the information of the
+         photons on the ground.
+         If the azimuth and zenith angles are provided, the Cherenkov photon's coordinates are
+         filled in the plane perpendicular to the incoming direction of the particle.
+
+        Parameters
+        ----------
+        photons: list
+            List of size M of numpy.array of size (N,8), where M is the number of telescopes in the
+            array, N is the number of photons that reached each telescope. The following information
+             of the Cherenkov photons on the ground are saved:
+             x: x position on the ground (CORSIKA coordinate system),
+             y: y position on the ground (CORSIKA coordinate system),
+             cx: direction cosinus in the x direction, i.e., the cosinus of the angle between the
+             incoming direction and the x axis,
+             cy: direction cosinus in the y direction, i.e., the cosinus of the angle between the
+             incoming direction and the y axis,
+             time: time of arrival of the photon in ns. The clock starts when the particle crosses
+             the top of the atmosphere (CORSIKA-defined) if `self.event_first_interaction_heights`
+             is positive or at first interaction if otherwise.
+             zem: altitude where the photon was generated in cm,
+             photons: number of photons associated to this bunch,
+             wavelength: the wavelength of the photons in nm.
```

Comment:
Do you actually get a photon wavelength of the bunch size is != 1?

Reply:
I am not sure I get the question. 
`photons` gives the number of photons in the bunch and `wavelength` represents the wavelength of all the photons in the bunch.

the specific corsika IACT file that I use in the tests was configured for bunch size 1, so to be honest I am not sure what would happen if this would be >1, I would have to test a different file.

Reply:
I opened the issue #472 to address this comment later and make sure we can handle different input configurations than the one used for tests here